tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -0,0 +1,767 @@
+{
+	uint8_t buf[] = {};
+	ssize_t ret;
+
+	ret = fixture->ase_cp->write(&fixture->conn, fixture->ase_cp, (void *)buf, 0, 0, 0);
+	zassert_true(ret < 0, "ase_cp_attr->write returned unexpected (err 0x%02x)",
+		     BT_GATT_ERR(ret));
+}
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_invalid_opcode)
+{
+	uint8_t buf[] = {
+		0x00,   /* Opcode */
+		0x01,   /* Number_of_ASEs */
+		0x01,   /* ASE_ID[0] */
+	};
+	uint8_t data_expected[] = {
+		0x00,   /* Opcode */
+		0xFF,   /* Number_of_ASEs */
+		0x00,   /* ASE_ID[0] */
+		0x01,   /* Response_Code[0] = Unsupported Opcode */
+		0x00,   /* Reason[0] */
+	};
+
+	bt_conn_index_fake.return_val = 0;
+
+	fixture->ase_cp->write(&fixture->conn, fixture->ase_cp, (void *)buf, sizeof(buf), 0, 0);
+
+	expect_single_call_bt_gatt_notify_cb(&fixture->conn, BT_UUID_ASCS_ASE_CP, fixture->ase_cp,
+					     data_expected, sizeof(data_expected));
+}
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_unsupported_opcode)
+{
+	uint8_t buf[] = {
+		0xFE,   /* Opcode */
+		0x01,   /* Number_of_ASEs */
+		0x01,   /* ASE_ID[0] */
+	};
+	uint8_t data_expected[] = {
+		0x00,   /* Opcode */
+		0xFF,   /* Number_of_ASEs */
+		0x00,   /* ASE_ID[0] */
+		0x01,   /* Response_Code[0] = Unsupported Opcode */
+		0x00,   /* Reason[0] */
+	};
+
+	bt_conn_index_fake.return_val = 0;
+
+	fixture->ase_cp->write(&fixture->conn, fixture->ase_cp, (void *)buf, sizeof(buf), 0, 0);
+
+	expect_single_call_bt_gatt_notify_cb(&fixture->conn, BT_UUID_ASCS_ASE_CP, fixture->ase_cp,
+					     data_expected, sizeof(data_expected));
+}
+
+/*
+ *  Test correctly formatted ASE Control Point 'Invalid Length' notification is sent
+ *
+ *  ASCS_v1.0; 5 ASE Control operations
+ *  "A client-initiated ASE Control operation shall be defined as an invalid length operation
+ *   if the Number_of_ASEs parameter value is less than 1"
+ *
+ *  Constraints:
+ *   - Number_of_ASEs is set to 0
+ *   - Config Codec operation parameter array is valid
+ *
+ *  Expected behaviour:
+ *   - "If the Response_Code value is 0x01 or 0x02, Number_of_ASEs shall be set to 0xFF."
+ *   - ASE Control Point notification is correctly formatted
+ */
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_codec_configure_invalid_length_1)
+{
+	uint8_t buf[] = {
+		0x01,           /* Opcode = Config Codec */
+		0x00,           /* Number_of_ASEs */
+		0x01,           /* ASE_ID[0] */
+		0x01,           /* Target_Latency[0] = Target low latency */
+		0x02,           /* Target_PHY[0] = LE 2M PHY */
+		0x06,           /* Codec_ID[0].Coding_Format = LC3 */
+		0x00, 0x00,     /* Codec_ID[0].Company_ID */
+		0x00, 0x00,     /* Codec_ID[0].Vendor_Specific_Codec_ID */
+		0x00,           /* Codec_Specific_Configuration_Length[0] */
+	};
+	uint8_t data_expected[] = {
+		0x01,           /* Opcode */
+		0xFF,           /* Number_of_ASEs */
+		0x00,           /* ASE_ID[0] */
+		0x02,           /* Response_Code[0] = Invalid Length */
```

Comment:
Is invalid length what we should expect if number of ASEs = 0?

Reply:
Yes, the ASE_ID is 0 because the Response_Code is 0x02, as defined in ASCS specification:
 
![image](https://user-images.githubusercontent.com/10126877/225237506-db61e88e-4f4e-47ad-8ffa-d00922336344.png)

Reply:
My comment was not for the ASE_ID, but the Invalid Length response code :)

Reply:
Not sure what you mean. In request, the ASE_ID is 0x01. In response ASE_ID is 0x00 as per spec

Reply:
The ASE_ID in `buf` (request) was 0x01. You were probably looking at response, where it's 0x00